### PR TITLE
add metric if service hits resource record is over the limit per record set

### PR DESCRIPTION
**Pull request by zhiyanfoo**
Tested on `elysium.us1.staging.dog` 

We see the metric when we artificially limited the max record set and increased the number of jukebox-server pods we ran.

https://ddstaging.datadoghq.com/dashboard/vr9-e2a-w5d/zhiyans-personal-dashboard?fullscreen_end_ts=1674598659949&fullscreen_paused=false&fullscreen_section=overview&fullscreen_start_ts=1674595059949&fullscreen_widget=3585874172741624&from_ts=1674595058310&to_ts=1674598658310&live=true

### CODE REVIEW

**Comment by zhiyanfoo** [comment]
Current dependencies on/for this PR:
* datadog-master-v2
  * **PR #15** <a href="https://app.graphite.dev/github/pr/DataDog/external-dns/15" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/DataDog/external-dns/15?utm_source=stack-comment).

File: provider/aws/aws.go [hunk 1]
```diff
@@ -687,8 +707,24 @@ func (p *AWSProvider) newChange(action string, ep *endpoint.Endpoint) (*route53.
 	return change, dualstack
 }
 
-func truncateEndpointTargetSubset(ep *endpoint.Endpoint) []string {
+// For the resourceRecordsOverLimitForRecordSet we only set it initially if it ever hits the
+// limit. Once it hits the limit we will keep track of the gauge, and if it ever falls below the
+// limit we set it to 0.
+func truncateEndpointTargetSubset(
```
Comment (LaureHalabi) [inline provider/aws/aws.go:713]: nit: can this function header stay on one line? So it's the same format as the other functions in the script.

File: provider/aws/aws.go [hunk 2]
```diff
@@ -126,8 +131,17 @@ var (
 		// Cloudfront
 		"cloudfront.net": "Z2FDTNDATAQYW2",
 	}
+
+	resourceRecordsOverLimitForRecordSet = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "resource_records_over_limit_for_record_set",
+	}, []string{"dns_name", "record_type", "set_identifier"})
+	gaugeSetPerResourceRecordLabel = map[recordSetIdentifier]struct{}{}
```
Comment (LaureHalabi) [inline provider/aws/aws.go:138]: For my own understanding, can you explain what these variables are for and how they're used?

Comment (zhiyanfoo) [reply provider/aws/aws.go:138]: Good question. I'll add it as a comment in the PR

**Review by LaureHalabi — APPROVED** [review]
LGTM, but another review from someone with external-dns might be helpful

**Pull request merged**